packages/docs/pages/analytics/index.mdx

```diff
@@ -0,0 +1,165 @@
+
+```jsx
+
+useEffect(() => {
+  /**
+   * Set custom metadata for segment analytics page event
+   * Data can be updated anytime during the runtime
+   */
+  setPageMetadata({
+    customPropertyOne: true,
+    customPropertyTwo: 'John Doe',
+    customDynamicProperty: history.location.pathname,
+  });
+}, [history.location.pathname]);
+
+```
+
+### Default page event data sample
+
+```json
+  "anonymousId": "a4b53248-c0ab-411d-bfba-625de10eac1b",
+  "context": {
+    "campaign": {},
+    "groupId": "11789772",
+    "ip": "66.187.232.127",
+    "library": {
+      "name": "analytics.js",
+      "version": "npm:next-1.38.0"
+    },
+    "locale": "en-US",
+    "page": {
+      "path": "/insights/vulnerability/cves",
+      "referrer": "",
+      "search": "?affecting=true&page=1&page_size=20&sort=-cvss_score",
+      "title": "CVEs - Vulnerability | Red Hat Insights",
+      "url": "https://gov.console.stage.redhat.com/insights/vulnerability/cves?affecting=true&page=1&page_size=20&sort=-cvss_score"
+    },
+    "userAgent": "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/101.0.4951.54 Safari/537.36"
+  },
+  "integrations": {},
+  "messageId": "ajs-next-4052c0f68ba89b1cf7805543de3c0cfe",
+  "originalTimestamp": "2022-07-07T09:05:07.998Z",
+  "properties": {
+    "isBeta": true,
+    "module": "vulnerability",
+    "path": "/insights/vulnerability/cves",
+    "referrer": "",
+    "search": "?affecting=true&page=1&page_size=20&sort=-cvss_score",
+    "title": "CVEs - Vulnerability | Red Hat Insights",
+    "url": "https://gov.console.stage.redhat.com/insights/vulnerability/cves?affecting=true&page=1&page_size=20&sort=-cvss_score"
+  },
+  "receivedAt": "2022-07-07T09:05:08.032Z",
+  "sentAt": "2022-07-07T09:05:07.998Z",
+  "timestamp": "2022-07-07T09:05:08.032Z",
+  "type": "page",
+  "userId": "51834776"
+```
+
+## Custom event tracking
+
+Chrome API also exposes the full `analytics` instance to allow custom event tracking.
+
+Checkout the <a href="https://github.com/RedHatInsights/insights-rbac-ui/pull/1188" target="_blank">RBAC UI example <ExternalIconAlt />.</a>
+
+### Example: tracking user click event
+
+```jsx
+import useChrome from '@redhat-cloud-services/frontend-components/useChrome';
+
+...
+
+const RemoveEntityButton = ({ entityId, handleRemove }) => {
+  const { analytics } = useChrome();
+
+  return (
+    <div>
+      <button onClick={() => {
+        analytics.track('remove-entity', {
+          entityId
+        });
+        handleRemove()
+      }}>Remove entity: {entityId}</button>
+    </div>
+  )
+}
+
+```
+
+## Creating custom UI module segment
+
+*By default, UI modules share the `Hybrid Cloud Console Dev` and `Hybrid Cloud Console` segments.*
+
+The segment analytics allows definig multiple sources/namespaces to separate (segment) analytics data. To create a new source for a module, it is necessary to provide a unique API key. It can be existing key or new key.
```

Comment:
```suggestion
The segment analytics allows defining multiple sources/namespaces to separate (segment) analytics data. To create a new source for a module, it is necessary to provide a unique API key. It can be an existing key or new key.
```